src/app/canvas/canvas/canvas.component.ts

```diff
@@ -43,68 +46,59 @@ export class CanvasComponent {
   onMousewheel(event: WheelEvent) {
     event.preventDefault();
     const zoomScale = 1 + (event.deltaY * -0.002);
-    const zoomPnt = {
-      x: event.clientX - this.getBounds().left,
-      y: event.clientY - this.getBounds().top
-    };
-    this.canvasService.zoom(zoomPnt, zoomScale);
+    const zoomPnt = new ViewportCoord(
```

Comment:
You can take the engineer out of Java-land, but can you take Java-land out of the engineer?

Reply:
(all jokes though, ES6 classes are just syntactic sugar)